### PR TITLE
move match_histogram from tranform to exposure

### LINKED ISSUE #3618: Move histogram matching from transform to exposure
We recently merged #3568, which added histogram matching to the `transform` module. @stefanv argued (and I agree) that it should go into `exposure` instead. Oops!

See https://github.com/scikit-image/scikit-image/pull/3568#pullrequestreview-186841972

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Closes #3618

Would be good to merge this for 0.16 for fix the mistake asap and reduce the burden for our users.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Before merging, we should make sure that the warnings are all caught in the test suite
https://travis-ci.org/scikit-image/scikit-image/jobs/574881051#L2689

**Comment by sciunto** [comment]
No warning in the raw log anymore. I fixed them.

File: skimage/transform/histogram_matching.py [hunk 1]
```diff
@@ -1,72 +1,25 @@
...
-        Image to match histogram of. Must have the same number of channels as
-        image.
-    multichannel : bool, optional
-        Apply the matching separately for each channel.
+if mh.__doc__ is not None:
+    match_histograms.__doc__ = mh.__doc__ + """
+    Warns
+    -----
+    Deprecated:
+        .. versionadded:: 0.16
```
Comment (hmaarrfk) [inline skimage/transform/histogram_matching.py]: nice touch.

**Comment by sciunto** [comment]
@jni would you have time to look at this one please? You posted the issue.

File: skimage/transform/histogram_matching.py (hunk 1 above)
Comment (jni) [reply skimage/transform/histogram_matching.py]: 👏

**Comment by sciunto** [comment]
Thanks @jni!

**Pull request merged**